webapp/components/topbar/dialogs/ActionDialog.js

```diff
@@ -38,8 +38,12 @@ class ActionDialog extends React.Component {
   performAction = () => {
     if (this.props.command) {
       if (this.props.isFormValid === undefined || this.props.isFormValid()) {
-        GEPPETTO.trigger(GEPPETTO.Events.Show_spinner, this.props.message);
-        this.props.pythonCall(this.props.command, this.props.args);
+        if (typeof this.props.callback !== 'undefined' || this.props.callback !== null) {
+          this.props.callback(this.props.command, this.props.args);
+        } else {
+          GEPPETTO.trigger(GEPPETTO.Events.Show_spinner, this.props.message);
```

Comment:
Legacy spinner trigger: the limitation of this approach is that parallel asynchronous operations may trigger the spinner and the first closer closes for all. I suggest to always use The specific action https://github.com/MetaCell/geppetto-meta/blob/master/geppetto.js/geppetto-client/src/common/actions/actions.js#L141 and specify a close action (in this case would be a receive python message I suppose). cc @ddelpiano know you were on it, what do you suggest here?

Reply:
@Muhaddatha just for reference, this has just been merged to development:

https://github.com/MetaCell/NetPyNE-UI/pull/466